packages/algob/test/internal/deployer.ts

```diff
@@ -677,3 +679,92 @@ describe("DeployerDeployMode", () => {
 		assert.deepEqual(res.assetDef, asaDef);
 	});
 });
+describe("helper functions", () => {
+	const env = mkEnv("network1");
+	const deployerCfg = new DeployerConfig(env, new AlgoOperatorDryRunImpl());
+	let deployer: DeployerDeployMode;
+	let alice: algosdk.Account;
+	let bob: algosdk.Account;
+
+	beforeEach(() => {
+		deployer = new DeployerDeployMode(deployerCfg);
+		alice = generateAccount();
+		bob = generateAccount();
+	});
+
+	it("Should return a transaction object based on provided execParams", async () => {
+		const execParams: AlgoTransferParam = {
+			type: TransactionType.TransferAlgo,
+			sign: SignType.SecretKey,
+			fromAccount: alice,
+			toAccountAddr: bob.addr,
+			amountMicroAlgos: 10000n,
+			payFlags: {},
+		};
+		const txnParams = mockSuggestedParam;
+		const transactions: Transaction[] = deployer.makeTx([execParams], txnParams);
+		assert.deepEqual(transactions[0].type, algosdk.TransactionType.pay);
+		assert.deepEqual(algosdk.encodeAddress(transactions[0].from.publicKey), alice.addr);
+		assert.deepEqual(algosdk.encodeAddress(transactions[0].to.publicKey), bob.addr);
+		assert.deepEqual(transactions[0].amount, 10000n);
+	});
+
+	it("Should sign a transaction and return a SignedTransaction object", async () => {
+		const execParams: AlgoTransferParam = {
+			type: TransactionType.TransferAlgo,
+			sign: SignType.SecretKey,
+			fromAccount: alice,
+			toAccountAddr: bob.addr,
+			amountMicroAlgos: 10000n,
+			payFlags: {},
+		};
+		const signature: wtypes.Sign = {
+			sign: SignType.SecretKey,
+			fromAccount: alice,
+		};
+		const txnParams = mockSuggestedParam;
+		const transactions: Transaction[] = deployer.makeTx([execParams], txnParams);
+		assert.doesNotThrow(() => {
+			deployer.signTx(transactions[0], signature);
+		});
+	});
+
+	it("Should return a SignedTransaction object based on ExecParams", async () => {
+		const execParams: AlgoTransferParam = {
+			type: TransactionType.TransferAlgo,
+			sign: SignType.SecretKey,
+			fromAccount: alice,
+			toAccountAddr: bob.addr,
+			amountMicroAlgos: 10000n,
+			payFlags: {},
+		};
+		const signature: wtypes.Sign = {
+			sign: SignType.SecretKey,
+			fromAccount: alice,
+		};
+		const txnParams = mockSuggestedParam;
+		assert.doesNotThrow(() => {
+			deployer.makeAndSignTx([execParams], txnParams, signature);
+		});
+	});
+
+	it("Should send a signed transaction and wait specified rounds for confirmation", async () => {
```

Comment:
same here

Reply:
the functions in there are asynchronous that's why we should use it

Reply:
However you are right. The async can be moved to the assert